<?xml version='1.0' encoding='UTF-8'?>
<root>
 <tabbed_widget name="Main Window" parent="main_window">
  <Tab containers="1" tab_name="tab1">
   <Container>
    <DockSplitter orientation="-" count="4" sizes="0.247551;0.243989;0.242208;0.266251">
     <DockSplitter orientation="|" count="3" sizes="0.333156;0.333687;0.333156">
      <DockArea name="...">
       <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
        <range left="27.700458" top="0.616471" right="66.198363" bottom="0.416471"/>
        <limitY/>
        <curve name="/planning/planning_validator/validation_status/distance_deviation" color="#1f77b4"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
        <range left="27.700458" top="0.350000" right="66.198363" bottom="0.150000"/>
        <limitY/>
        <curve name="/planning/planning_validator/validation_status/velocity_deviation" color="#d62728"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
        <range left="27.700458" top="0.100003" right="66.198363" bottom="-0.099997"/>
        <limitY/>
        <curve name="/planning/planning_validator/validation_status/max_curvature" color="#d62728"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter orientation="|" count="3" sizes="0.333156;0.333687;0.333156">
      <DockArea name="...">
       <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
        <range left="27.700458" top="1.000000" right="66.198363" bottom="1.000000"/>
        <limitY/>
        <curve name="/planning/planning_validator/validation_status/max_interval_distance" color="#1ac938"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
        <range left="27.700458" top="0.000016" right="66.198363" bottom="0.000016"/>
        <limitY/>
        <curve name="/planning/planning_validator/validation_status/max_lateral_acc" color="#ff7f0e"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
        <range left="27.700458" top="1.118624" right="66.198363" bottom="0.918624"/>
        <limitY/>
        <curve name="/planning/planning_validator/validation_status/max_longitudinal_acc" color="#f14cc1"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter orientation="|" count="3" sizes="0.333156;0.333687;0.333156">
      <DockArea name="...">
       <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
        <range left="27.700458" top="0.100003" right="66.198363" bottom="-0.099997"/>
        <limitY/>
        <curve name="/planning/planning_validator/validation_status/max_relative_angle" color="#9467bd"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
        <range left="27.700458" top="0.100007" right="66.198363" bottom="-0.099993"/>
        <limitY/>
        <curve name="/planning/planning_validator/validation_status/max_steering" color="#17becf"/>
       </plot>
      </DockArea>
      <DockArea name="...">
       <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
        <range left="27.700458" top="0.000011" right="66.198363" bottom="0.000010"/>
        <limitY/>
        <curve name="/planning/planning_validator/validation_status/max_steering_rate" color="#bcbd22"/>
       </plot>
      </DockArea>
     </DockSplitter>
     <DockSplitter orientation="|" count="2" sizes="0.33245;0.66755">
      <DockArea name="...">
       <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
        <range left="27.700458" top="0.100000" right="66.198363" bottom="-0.100000"/>
        <limitY/>
        <curve name="/planning/planning_validator/validation_status/min_longitudinal_acc" color="#1f77b4"/>
       </plot>
      </DockArea>
      <DockSplitter orientation="|" count="2" sizes="0.5;0.5">
       <DockArea name="...">
        <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
         <range left="27.700458" top="1.100000" right="66.198363" bottom="0.900000"/>
         <limitY/>
         <curve name="/planning/planning_validator/validation_status/is_valid_curvature" color="#1ac938"/>
         <curve name="/planning/planning_validator/validation_status/is_valid_distance_deviation" color="#ff7f0e"/>
         <curve name="/planning/planning_validator/validation_status/is_valid_finite_value" color="#f14cc1"/>
         <curve name="/planning/planning_validator/validation_status/is_valid_interval" color="#9467bd"/>
         <curve name="/planning/planning_validator/validation_status/is_valid_lateral_acc" color="#17becf"/>
        </plot>
       </DockArea>
       <DockArea name="...">
        <plot style="Lines" flip_y="false" mode="TimeSeries" flip_x="false">
         <range left="27.700458" top="1.100000" right="66.198363" bottom="0.900000"/>
         <limitY/>
         <curve name="/planning/planning_validator/validation_status/is_valid_longitudinal_max_acc" color="#bcbd22"/>
         <curve name="/planning/planning_validator/validation_status/is_valid_longitudinal_min_acc" color="#1f77b4"/>
         <curve name="/planning/planning_validator/validation_status/is_valid_relative_angle" color="#d62728"/>
         <curve name="/planning/planning_validator/validation_status/is_valid_steering" color="#1ac938"/>
         <curve name="/planning/planning_validator/validation_status/is_valid_steering_rate" color="#ff7f0e"/>
         <curve name="/planning/planning_validator/validation_status/is_valid_velocity_deviation" color="#f14cc1"/>
        </plot>
       </DockArea>
      </DockSplitter>
     </DockSplitter>
    </DockSplitter>
   </Container>
  </Tab>
  <currentTabIndex index="0"/>
 </tabbed_widget>
 <use_relative_time_offset enabled="1"/>
 <!-- - - - - - - - - - - - - - - -->
 <!-- - - - - - - - - - - - - - - -->
 <Plugins>
  <plugin ID="DataLoad CSV">
   <default time_axis="" delimiter="0"/>
  </plugin>
  <plugin ID="DataLoad MCAP"/>
  <plugin ID="DataLoad ROS2 bags">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="true"/>
   <max_array_size value="100"/>
  </plugin>
  <plugin ID="DataLoad ULog"/>
  <plugin ID="ROS2 Topic Subscriber">
   <use_header_stamp value="false"/>
   <discard_large_arrays value="false"/>
   <max_array_size value="true"/>
   <selected_topics>
    <topic name="/planning/planning_validator/validation_status"/>
   </selected_topics>
  </plugin>
  <plugin ID="UDP Server"/>
  <plugin ID="WebSocket Server"/>
  <plugin ID="ZMQ Subscriber"/>
  <plugin ID="Fast Fourier Transform"/>
  <plugin ID="Quaternion to RPY"/>
  <plugin ID="Reactive Script Editor">
   <library code="--[[ Helper function to create a series from arrays&#xa;&#xa; new_series: a series previously created with ScatterXY.new(name)&#xa; prefix:     prefix of the timeseries, before the index of the array&#xa; suffix_X:   suffix to complete the name of the series containing the X value. If [nil], use the index of the array.&#xa; suffix_Y:   suffix to complete the name of the series containing the Y value&#xa; timestamp:   usually the tracker_time variable&#xa;              &#xa; Example:&#xa; &#xa; Assuming we have multiple series in the form:&#xa; &#xa;   /trajectory/node.{X}/position/x&#xa;   /trajectory/node.{X}/position/y&#xa;   &#xa; where {N} is the index of the array (integer). We can create a reactive series from the array with:&#xa; &#xa;   new_series = ScatterXY.new(&quot;my_trajectory&quot;) &#xa;   CreateSeriesFromArray( new_series, &quot;/trajectory/node&quot;, &quot;position/x&quot;, &quot;position/y&quot;, tracker_time );&#xa;--]]&#xa;&#xa;function CreateSeriesFromArray( new_series, prefix, suffix_X, suffix_Y, timestamp )&#xa;  &#xa;  --- clear previous values&#xa;  new_series:clear()&#xa;  &#xa;  --- Append points to new_series&#xa;  index = 0&#xa;  while(true) do&#xa;&#xa;    x = index;&#xa;    -- if not nil, get the X coordinate from a series&#xa;    if suffix_X ~= nil then &#xa;      series_x = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_X) )&#xa;      if series_x == nil then break end&#xa;      x = series_x:atTime(timestamp)&#x9; &#xa;    end&#xa;    &#xa;    series_y = TimeseriesView.find( string.format( &quot;%s.%d/%s&quot;, prefix, index, suffix_Y) )&#xa;    if series_y == nil then break end &#xa;    y = series_y:atTime(timestamp)&#xa;    &#xa;    new_series:push_back(x,y)&#xa;    index = index+1&#xa;  end&#xa;end&#xa;&#xa;--[[ Similar to the built-in function GetSeriesNames(), but select only the names with a give prefix. --]]&#xa;&#xa;function GetSeriesNamesByPrefix(prefix)&#xa;  -- GetSeriesNames(9 is a built-in function&#xa;  all_names = GetSeriesNames()&#xa;  filtered_names = {}&#xa;  for i, name in ipairs(all_names)  do&#xa;    -- check the prefix&#xa;    if name:find(prefix, 1, #prefix) then&#xa;      table.insert(filtered_names, name);&#xa;    end&#xa;  end&#xa;  return filtered_names&#xa;end&#xa;&#xa;--[[ Modify an existing series, applying offsets to all their X and Y values&#xa;&#xa; series: an existing timeseries, obtained with TimeseriesView.find(name)&#xa; delta_x: offset to apply to each x value&#xa; delta_y: offset to apply to each y value &#xa;  &#xa;--]]&#xa;&#xa;function ApplyOffsetInPlace(series, delta_x, delta_y)&#xa;  -- use C++ indeces, not Lua indeces&#xa;  for index=0, series:size()-1 do&#xa;    x,y = series:at(index)&#xa;    series:set(index, x + delta_x, y + delta_y)&#xa;  end&#xa;end&#xa;"/>
   <scripts/>
  </plugin>
  <plugin ID="CSV Exporter"/>
  <plugin ID="ROS2 Topic Re-Publisher"/>
 </Plugins>
 <!-- - - - - - - - - - - - - - - -->
 <previouslyLoaded_Datafiles/>
 <previouslyLoaded_Streamer name="ROS2 Topic Subscriber"/>
 <!-- - - - - - - - - - - - - - - -->
 <customMathEquations/>
 <snippets/>
 <!-- - - - - - - - - - - - - - - -->
</root>
